promptsource/seqio_tasks/tasks.py

```diff
@@ -148,16 +147,51 @@ def add_task(dataset_name, subset_name, template_name, task_name=None, split_map
         )
 
 
-all_templates = promptsource.templates.TemplateCollection()
+train_sets: List[Dict] = []
+eval_sets: List[Dict] = []
+do_train: List[Tuple[str, Optional[str]]] = []
+do_eval: List[Tuple[str, Optional[str]]] = []
+experiment_path = pkg_resources.resource_filename(__name__, "experiment_D4.csv")
+with open(experiment_path) as exp_file:
+    reader = csv.DictReader(exp_file)
+    for row in reader:
+        if row["skip"]:
+            continue
+        if row["subset"] == "":
+            row["subset"] = None  # to match promptsource.Template object
+        if row["do_train"] == "TRUE":
+            train_sets.append(row)
+            do_train.append((row["HF_name"], row["subset"]))
+        if row["do_eval"] == "TRUE":
+            eval_sets.append(row)
+            do_eval.append((row["HF_name"], row["subset"]))
+
+train_or_eval = do_train + do_eval
+print(f"Number of training datasets = {len(train_sets)}")
+print(f"Number of evaluation datasets = {len(eval_sets)}")
```

Comment:
Sort of a nitpick but can you use logging for this, or remove?

Reply:
It also looks like train_sets and eval_sets are only populated to print their length. I think you can just log the length of do_train/do_eval and not crete those lists.